### DISCUSSION TITLE
New GUI with export to 3dplot LS-Dyna

### DISCUSSION THREAD

**Discussion by ZdendaH** [OpenRadioss/OpenRadioss#2819] [Q&A] [answered]
Dear OpenRadioss Comunity,,

Can I ask for help to solve this issue? I wanted to try the new GUI which was released on 4th July 2024 (mk4) and it includes the AnimTo3dplot function.
I always get these message when it runs. I am using OpenRadioss under Windows 11

<img width="800" alt="image" src="https://github.com/user-attachments/assets/e8cfa380-86ae-4829-8b2b-09137d62510a">

Anim-d3plot option selected, Converting Anim Files to d3plot
------------------------------------------------------------
python: can't open file 'C:\\01_OpenRadioss\\OpenRadioss\\exec\\d3convert.py': [Errno 2] No such file or directory
---------------------------------------
Anim file conversion to d3plot complete

I can not find the file d3convert.py according to path from message on the picture.

Thank you for your advice.
Best Regards.

**Comment by andprad74** [comment, marked as answer]
Hello
i don't know if you have solve the issue but i solved with the following steps :

1) Extract the zip file, in attachment, in the "....\openradioos\exec\" directory please see the picture for the final result.
2) Download Python from "https://www.python.org/downloads/windows/" and install it with the PATH option flagged. See the picture.
3) Reboot the system. With Python is installed also Pip that is necessary to install the requirements Python Lasso, NumPy, Rich, typing and Tqdm.
4) After the reboot. it's possible to install the requirements. Type cmd in the windows find in order to open a DOS windows. See the picture
5) Install the lasso python by tiping in the DOs windows the following command:
python -m pip install lasso-python
6) In the same DOS windows install numpy with the following command;
pip install numpy
7) In the same DOS windows install rich with the following command ;
pip install numpy
8) In the same DOS windows install tqdm with the following comman ;
pip install tqdm 

Test the result ! I hope that the conversation will work.
by.
Andrea
[d3convert_and_vortex_radioss.zip](https://github.com/user-attachments/files/16652313/d3convert_and_vortex_radioss.zip)
![exec_directory](https://github.com/user-attachments/assets/5674b29d-d47c-4442-8897-e64d4d00dba4)

![python_with_the_PATH_option](https://github.com/user-attachments/assets/7d629502-8f7d-486d-ba2b-123f5c32864f)
![cmd_in_order-to_open_a_DOS_windows](https://github.com/user-attachments/assets/906bbdc8-8fd4-4bd0-88de-9af9303e3930)
![lasso_python](https://github.com/user-attachments/assets/ce244b6c-f0fa-42ae-9ca2-54ebe2af2781)

**Reply by elequiniou** [reply]
Thanks a lot Andrea for your step by step successful explanation. Developers also recommend to use latest Python like the one you use here. I propose to mark the discussion as answered.

**Reply by andprad74** [reply]
You welcome!
Yes, I agree with you, the discussion can be considered as  answered.
Thanks a lot for everything.
Andrea.

**Reply by ZdendaH** [reply]
Thank you very much for answer. It works. The thread is answered.

**Comment by dgrosenick** [comment]
I'm not sure if this is where to ask the question..
I have not been successful in getting the openradioss gui to display the d3plot checkbox. (windows 10 and 11)
I've installed python, then a clean complete pip install of lasso-python. I've placed the latest "vortex-radioss-official" code into c:\openradioss\exec\      (I've also tried renaming the folder to "vortex-radioss"  instead of "vortex-radioss-official)
I've got the latest openradioss gui (downloaded 12/18/24)
Where is vortex radioss code supposed to be placed that the gui code will find it (I'm using openradioss_gui.vbs)
Any help or direction where to ask would be appreciated.
Thanks
Dale G

**Reply by PaulAltair** [reply]
The vortex radioss tool is a python module and needs to be installed in your python installation folders rather than in openradioss/exec (from command line in vortex radioss main folder, you may run` setup.py install` (may need to be admin for this as setup is an older way of distributing python modules) or you can put the vortex-radioss folder (containing animtod3plot) directly in your 'Lib' folder of your python installation

**Reply by bernoertel** [reply]
I want to try the new GUI too which was released on 4th July 2024 (mk4) and which includes the AnimTo3dplot function. In my GUI I have only  Anim-vtk  and  TH-csv - check boxes.  Where can I download the new GUI  ??

**Reply by dgrosenick** [reply]
Paul, thanks for the help! I've got it going.
Some tips for anyone else trying.

install lasso-python   (  python -m pip install lasso-python   )
To get lasso-python to install clean I had to first install visual studio c++ tools  (  https://visualstudio.microsoft.com/visual-cpp-build-tools/  )
Without that numpy failed to install  (error below)_
-------------
Project name: NumPy
      Project version: 1.26.4
      WARNING: Failed to activate VS environment: Could not find C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe
------------

 running "setup.py install' did the trick

**Reply by dgrosenick** [reply]
When it first didn't install the button for me I converted manually. I made a python script with the path in the script.
----------
from vortex_radioss.animtod3plot.Anim_to_D3plot import readAndConvert
readAndConvert("C:\\\FEA\\\122224_constrained\\\dev1")
-----------

This worked great. Thanks again !  
I'm hoping for thickness, thinning and fld to be implemented in this translator !

**Comment by bernoertel** [comment]
Hallo ZdendaH and andprad74, 
I have all done as described, downloaded files, successful installed python and then lasso-python, numpy, rich, tqdm as described.  But I do not have the OpenRadioss-GUI with a 3dplot-checkbox inside.  Where can I download this new GUI  ??  Can you send me a download-link please  ?

When I try to use d3convert.py manually from exec-directory  
G:\OpenRadioss\exec>python d3convert.py C:/Users/Bern-Workstation/OneDrive/Desktop/FEM_Projekt/Bsp_Ball_Plate/ballA*  then I get
No files found..
C:/Folder/ModelA00* and so on ,
although the result-files  ballA001 - ballA101  are exactly located there.

**Reply by dgrosenick** [reply]
I also could not get d3convert.py to run from the command line. I believe the syntax for specifying the path might be very specific with python. 
That's why I tested it out with a pyton script where the path was in the script. I found that in the script the complete path had to be in double quotes (my guess is to escape the colon character) and all back-slash had to be double (I think that's the escape character like in unix shells and perl) Then all of that was in parenthesis.
See the example in my comment from a couple days ago. Place the 2 lines in a flat text file with the file extension .py
Then run in a command prompt   --    py *.py
If that runs, all the files are installed where they are supposed to be. What surprised me was several days later the button showed up in the openradioss gui !

**Reply by andprad74** [reply]
Hello Bernoertel and Dgrosenick,
please see the following link
https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/58785793/Latest+gui+scripts
https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/426409985/Archive+of+gui+scripts

Since August something is changed and in that period the windows scripts was  the win_scripts_mk4.zip,
that you can find in attachment.
I hope will help you.
By
Andrea


[win_scripts_mk4.zip](https://github.com/user-attachments/files/18308231/win_scripts_mk4.zip)

**Reply by bernoertel** [reply]
Thank you andprad74, unpacking your win_scripts_mk4.zip and then using that folder as   "openradioss_gui"  brings a new GUI with the wished button for d3plot inside.  startscript is then  "openradioss_run_script_for_windows_mpi_ps.exe".

**Reply by bernoertel** [reply]
Hello andprat74 and  other members.
for Windows the d3plot-Button did appear after using  "win_scripts_mk4.zip".  Now I need the same for LINUX.  Does somebody know, if d3plot  works under LINUX too and if YES --- can I get the necessary files for download  ??

**Reply by kalpanadevi2011** [reply]
Hello bernoertel, 
How to use the script to get the latest GUI to run, "openradioss_run_script_for_windows_mpi_ps.exe" does not working , Kindly guide eon this.

**Comment by OlivierWienholtz** [comment]
Hello,
Please note : the GUI is now shipped with the OpenRadioss releases.
Download latest releases at : https://github.com/OpenRadioss/OpenRadioss/releases
Note as well : OpenRadioss GUI source code is now in : https://github.com/OpenRadioss/Tools

--
Olivier
